accepted/00078-maintenance-windows-web-calendar.md

```diff
@@ -0,0 +1,181 @@
+
+### What use cases does it support?
+
+When choosing a maintenance schedule from a maintenance calendar the user needs to know the details of the maintenance windows contained in the calendar.
+
+Typical workflow when creating new maintenance windows:
+
+1. Head to `Schedule` -> `Maintenance Windows` -> `Calendars` -> `Create`
+2. Follow the steps to create a new calendar
+3. Go to `Schedules` to create a new maintenance schedule
+4. Use the information provided in the created calendars details view to pick a maintenance window
+5. Assign systems to the newly created maintenance schedule
+
+### What is the expected outcome?
+
+Replace the plain ical text in the calendar details and creation menu with a visual representation of the maintenance windows contained in the uploaded ical file.
+Use the created calendar view to also display information about a maintenance window in the maintenance schedule details and creation menu. 
+
+# Detailed design
+[design]: #detailed-design
+
+### UI
+
+Following example screenshot should help to provide a basic idea of how the implementation should look like. (created using react-big-calendar)
+
+Month view:
+![Month view](images/00078-month-view.png)
+
+Day view: 
+![Day view](images/00078-day-view.png)
+
+Events belonging to different Maintenance Window should have a different color
+
+To prevent massive scrolling needed within the calendar view `Back` and `Next` buttons
+should navigate to the previous/next months/days that have at least one maintenance window.
+
+The calendar should have a `read-only` mode to prevent manipulation of displayed events but also support it for potential future use.
+
+### Getting data from the backend
+
+A new function to calculate the maintenance windows contained in a maintenance Calendar needs to be implemented. It looks like `calculateUpcomingMaintenanceWindows` in `IcalUtils` can be reused/adapted in this case.
+
+I see 3 ways of calculating the maintenance windows displayed in the calendar
+1. Stick to the current way and calculate the next x (e.g. 10) maintenance windows
+2. Calculate all the maintenance windows within a given period (e.g. one year) and limit the months displayed in the calendar.
+3. Make a async call to the backend to retrieve all the maintenance windows of a month when clicking the `Back` and `Next` buttons. (basically server side pagination)
+
+From my point of view 1 is out. 3 would be the preferred way, but 2 is faster to implement and probably good enough.
+
+# Alternatives
+[alternatives]: #alternatives
+
+There seem to be a couple of solutions potentially suited for our use case:
+
+1. [Full Calendar](https://github.com/fullcalendar/fullcalendar)
+2. [React Big Calendar](https://github.com/jquense/react-big-calendar)
+3. [TOAST UI Calendar for React](https://github.com/nhn/toast-ui.react-calendar)
```

Comment:
And now the million-dollar question: as a designer and given the evidence you've accumulated out of the aspects you've evaluated, which solution would you pick - aka which criteria do you judge most important and which of the above projects satisfies them best?

That choice, and especially its motivation, could well represent the biggest value of this RFC!  

As a reviewer of course I have my idea but would like to hear your first, without influencing it too much. In the end, it will be you and your colleagues implementing and maintaining this, so your point of view is surely the richest.

Once the RFC contains the choice, if anybody has any concerns over the particular pick, the criteria or the research he/she can point out any doubt areas so that an informed choice is ultimately made.

Note that there will always be some risks and unknowns - there is no correct answer strictly speaking - unless we complete a full implementation of all possible options and run them all for a few years, we will never have perfect information! But that's fine, and that's why we like having research sessions, RFCs and constructive design conversations - to reduce some risks via an appropriately sized upfront time investment.

That's why we need (to form) good designers, that's what designs are for.

:+1: for all the work done so far

Reply:
Thanks, @parlt91. The comparison is really helpful